src/crypto/qat/qcccrypto.h

```diff
@@ -166,11 +169,53 @@ class QccCrypto {
      * associated callbacks. For synchronous operation (like this one), QAT
      * library creates an internal callback for the operation.
      */
-    static void* crypt_thread(void* entry);
-    CpaStatus QccCyStartPoll(int entry);
-    void poll_instance(int entry);
+    void poll_instances(void);
+
+    bool symPerformOp(int avail_inst,
+                      CpaCySymSessionCtx sessionCtx,
+                      const Cpa8U *pSrc,
+                      Cpa8U *pDst,
+                      Cpa32U size,
+                      Cpa8U *pIv,
+                      Cpa32U ivLen,
+                      optional_yield y);
+
+    CpaStatus initSession(CpaInstanceHandle cyInstHandle,
+                          CpaCySymSessionCtx *sessionCtx,
+                          Cpa8U *pCipherKey,
+                          CpaCySymCipherDirection cipherDirection);
+
+    CpaStatus updateSession(CpaCySymSessionCtx sessionCtx,
+                            Cpa8U *pCipherKey,
+                            CpaCySymCipherDirection cipherDirection);
+
+
+};
 
-    pthread_t *cypollthreads;
-    static const size_t qcc_sleep_duration = 2;
+class QatCrypto {
+  QccCrypto *crypto;
```

Comment:
it doesn't look like this pointer is used

Reply:
> it doesn't look like this pointer is used

Yes, it isn't used. I have removed it.